subsys/modem/modem_chat.c

```diff
@@ -78,6 +78,14 @@ static void modem_chat_script_stop(struct modem_chat *chat, enum modem_chat_scri
 		chat->script->callback(chat, result, chat->user_data);
 	}
 
+	/* Clear parse_match in case it is stored in the script being stopped */
+	if ((chat->parse_match) &&
```

Comment:
As `parse_match` does not seem be a boolean, then better would be `chat->parse_match != NULL`. We have been changing these to be like this in the code because of some Misra rule.

Reply:
Changed all occurrences (2) of `!foo` to `foo == NULL` :)